### PR TITLE
Rename 'tag' to 'branch_id'

### PR DESCRIPTION
**Pull request by talex5**
Using 'tag' to mean branch name is really confusing, because Git uses 'tag' to mean a named pointer that is *not* a branch.

BC.tag is now BC.name and BC.branch is now BC.head_ref.

The remote HTTP protocol still uses "tag".

This API change will break most users of Irmin.

### CODE REVIEW

**Comment by talex5** [comment]
Travis is now passing, since I updated it to test against an updated version of irmin-indexeddb.

**Comment by talex5** [comment]
Thinking further about this, I think renaming the internal "Tag store" to "Branch store" wasn't quite right (although using `branch` in the `BC` interface is good). I think I'll change it instead to "Ref store". That's general enough to cover actual tags if we add them later, and avoids confusion between a BC branch (which you can write files to) and a Ref.t (which just maps branch IDs to commit IDs). I'll update the patch today.

**Comment by talex5** [comment]
Updated to use `Ref` for references.

Examples changes to API:

    -  Store.merge_tag_exn (fmt t "Merging with %s" branch) branch >>= fun () ->
    +  Store.merge_branch_exn (fmt t "Merging with %s" branch) branch >>= fun () ->

and

    -  val tags: t -> tag list Lwt.t
    -  (** The list of all persistent branch names.
    +  val branches: t -> branch_id list Lwt.t
    +  (** The list of all persistent branch names.

**Comment by samoht** [comment]
Thanks! This all looks good. Just wondering why you choose `of_branch_id` and type instead of `of_branch` and `type branch_id` instead of `type branch`?

**Comment by talex5** [comment]
I want to distinguish between a branch ID (a string) and a branch (something you can read files from). In particular, I might later split out the branch-related functions of BC into a Branch module. Then `Branch.create` would take a branch ID and return a branch, so these concepts need to be kept separate.

**Comment by samoht** [comment]
I see, that makes sense. Merging now.

File: CHANGES.md [hunk 1]
```diff
@@ -8,6 +8,9 @@
   same offset in the `Cstruct.t` value, which is obviously not
   always the case. Apply the same fix to `Hash.SHA1.compare` and
   `Hash.SHA1.hash`.
+* Renamed "tag" to "branch" in the API, as "tag" is confusing for Git
+  users. `BC.tag` is now `BC.name` and `BC.branch` is now `BC.head_ref`.
+  Note: The remote HTTP protocol still uses "tag".
```
Comment (samoht) [inline CHANGES.md]: Can you also say that you rename `BC.of_tag` into `BC.of_branch_id`?

**Comment by samoht** [comment]
Actually, can you update the README to tell about the `Tag` to `Ref` renaming as well?

**Comment by talex5** [comment]
Done. Should be safe to merge, as only the README was updated.

**Pull request merged**